core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -702,10 +702,10 @@ public final void post(BuildListener listener) throws Exception {
                 post2(listener);
             } finally {
                 // update the culprit list
-                HashSet<String> r = new HashSet<>();
+                SortedSet<String> r = new TreeSet<>();
                 for (User u : getCulprits())
                     r.add(u.getId());
-                culprits = ImmutableSortedSet.copyOf(r);
+                culprits = Collections.unmodifiableSet(r);
```

Comment:
`r` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.